In [1]:
import pandas as pd

In [2]:
data = pd.read_table('daily_weather.tsv')

In [4]:
corrs = data.corr()

In [5]:
corrs['no_casual_riders']

weekday             0.125130
season_code         0.198099
is_holiday          0.029720
is_work_day        -0.539919
weather_code       -0.223149
temp                0.542253
subjective_temp     0.543917
humidity           -0.068902
windspeed          -0.158371
no_casual_riders    1.000000
no_reg_riders       0.274984
total_riders        0.643305
Name: no_casual_riders, dtype: float64

In [6]:
corrs['no_reg_riders']

weekday             0.129315
season_code         0.478133
is_holiday         -0.164190
is_work_day         0.437003
weather_code       -0.315485
temp                0.607425
subjective_temp     0.614727
humidity           -0.074875
windspeed          -0.265985
no_casual_riders    0.274984
no_reg_riders       1.000000
total_riders        0.912994
Name: no_reg_riders, dtype: float64

The most interesting out of the above is the is_work_day correlations: -.54 for casual riders and .43 for regular riders. This means less casual and more regular use the bikes on workdays vs weekends. There is no relationship between holidays and casual riders, but there is a slightly negative correlation for regular riders- likely because most are riding to work, and have that day off. Humidity never has an effect on riding. The total number of riders is much more dependent on the number of regular riders for that day than for the number of casual riders. 